twoliter/src/tools.rs

```diff
@@ -0,0 +1,138 @@
+use crate::common::fs;
+use crate::tools_hash::TOOLS_HASH;
+use anyhow::{Context, Result};
+use flate2::read::GzDecoder;
+use log::{debug, trace, warn};
+use std::path::Path;
+use tar::Archive;
+
+const TARBALL_DATA: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/tools.tar.gz"));
+
+pub(crate) async fn install_tools(tools_dir: impl AsRef<Path>, force: bool) -> Result<()> {
+    let tools_dir = tools_dir.as_ref();
+    let sentinel_filepath = tools_dir.join("installed");
```

Comment:
nit: could be a hidden file, which tends to imply the "sentinel" semantic:
```suggestion
    let sentinel_filepath = tools_dir.join(".installed");
```